### ISSUE TITLE
Idle timeout kills the main thread

### ISSUE THREAD

**Issue by toblux** [tonyg/racket-rfc6455#7]
The following program illustrates the issue:

```
#lang racket

(require net/url net/rfc6455)

;; Change idle timeout from 300 (the default) to 3 seconds.
(ws-idle-timeout 3)

;; Connect to any WebSocket server.
(define connection (ws-connect (string->url "wss://echo.websocket.org/")))

;; Wait 3 seconds and the main thread gets killed.
```

PR https://github.com/tonyg/racket-rfc6455/pull/6 fixes it.

**Comment by toblux** [comment]
@tonyg Any update on this?

The error is still reproducible after your latest changes. Did you have a chance to take a look at my PR? I've been using it locally without any problems for a while now.

My example above even kills the main thread when it's run in a new thread using `(thread thunk)`.

**Comment by tonyg** [comment]
Thanks for the reminder @toblux. I can reproduce the issue, but it is weirdly nondeterministic. That's probably not important though - now that I can reproduce it at least sometimes, I can try out your patch!

**Comment by tonyg** [comment]
OK, thanks for the patch. I'm not sure it has changed anything for me, but it shouldn't do any harm, and it might fix things for you, so... it's merged in #6. Please let me know whether things are resolved for you now!

**Comment by toblux** [comment]
Thanks, it works perfectly now.
